Amplify/Categories/DataStore/Model/Internal/ModelProvider.swift

```diff
@@ -0,0 +1,82 @@
+//
```

Comment:
Starting to tackle these, for this one specifically i got the error `Internal enum cannot be declared '@_spi' because only public and open declarations can be '@_spi'`.

Reply:
i was able to to add `@_spi` for `LazyReference.state` -> `LazyReference._state`, which makes sense as developer has an instance of LazyReference. Couldn't do this for `_LazyReferenceValueState` because of the chain: `_LazyReferenceValue` holds a `_LazyReferenceValueState` and `LazyReference` conforms to `_LazyReferenceValue`, causes the issue where LazyReference also needs to be `@_spi`.